inductiva/resources/machines.py

```diff
@@ -0,0 +1,107 @@
+"""Containes Machines class to manage resources."""
+
+from absl import logging
+from typing import Optional
+from uuid import UUID
+import uuid
+import inductiva
+from inductiva import api
+
+from inductiva.client import ApiClient, ApiException
+from inductiva.client.apis.tags.instance_api import InstanceApi
+from inductiva.client.model.instance import Instance
+from inductiva.client.model.instance_group import InstanceGroup
+
+
+class Machines():
+    """Class to Google Cloud manage resources."""
+
+    def __init__(
+        self,
+        machine_type: str,
+        executer: str,
+        num_machines: int = 1,
+        spot: bool = False,
+        disk_size_gb: int = 30,
+        label: str = None,
+        resource_pool_id: Optional[UUID] = None,
+    ) -> None:
+        """Initialize the Machines class.
+
+        Args:
+            machine_type: The type of GC machine to launch. Ex: "e2-medium".
+            executer: The type of executer to launch. Ex: "gromacs", "openfoam".
+            num_machines: The number of machines to launch.
+            spot: Whether to use spot instances.
+            disk_size_gb: The size of the disk in GB. (min. 30 GB)
+            label: The label to assign to the machine group.
+            resource_pool_id: The resource pool ID to use. A new resource pool
+              id can be created using the
+              inductiva.resources.create_resource_pool() method.
+        """
+        self.machine_type = machine_type
+        self.num_machines = num_machines
+        self.spot = spot
+        self.disk_size_gb = disk_size_gb
+        self.label = label
+        self.executer = executer
+        self.resource_pool_id = resource_pool_id
+        self.name = self._generate_instance_name()
+
+        self.api_config = api.validate_api_key(inductiva.api_key)
+
+    def start(self):
```

Comment:
should this method return something?

Reply:
in principle no, the endpoint returns the instance name , but since the users don't define the name of the instances, I didn't think it was necessary to return the name.